src/Symfony/Component/Mime/Part/Multipart/FormDataPart.php

```diff
@@ -75,6 +75,10 @@ private function prepareFields(array $fields): array
                 return;
             }
 
+            if (!\is_string($item) && !$item instanceof TextPart) {
```

Comment:
Maybe we can move this check to the constructor directly? That would be more consistent as we already check some of that there.

Reply:
I also considered this, but in my opinion it has two disadvantages:
1. We have to traverse through the array recursively. In `prepareFields` this is already the case, but in the constructor we only traverse at top-level.
2. We don't have the field name in form of "foo[qux][quux]" in the constructor. Either we cannot include it in the error message, or we have to generate it redundantly.

Also we already have a check (regarding the form field values with integer keys) inside `prepareFields`, so I don't feel like adding another check there affects the consistency much.

How about we remove the check in the constructor altogether? All the checks could be done inside `prepareFields` and the constructor would look like this:
```
 public function __construct(array $fields = [])
    {
        parent::__construct();

        $this->fields = $fields;

        // HTTP does not support \r\n in header values
        $this->getHeaders()->setMaxLineLength(\PHP_INT_MAX);
    }
```

Reply:
Having all the checks in prepareFields works for me.